### DISCUSSION TITLE
Support private github repositories self-hosted/poetry use cases

### DISCUSSION THREAD

**Discussion by yetanotherion** [renovatebot/renovate#22136] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

To fix the error at https://github.com/renovatebot/github-action/issues/738
We could use one of the tricks discussed at
https://github.com/orgs/python-poetry/discussions/3794

```
git config --global url."https://<renovate_token>".insteadOf "ssh://git"
```
before running [poetry update commands](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/poetry/artifacts.ts#L182), 
by using an env variable, or if any url points to ssh/github.

WDYT?

**Comment by viceice** [comment]
that can work, we use that for other managers too

**Reply by rarkins** [reply]
I really would like us to have a generic set of rewrites we reuse across managers

**Reply by yetanotherion** [reply]
So we managed to make it work with
```
GIT_CONFIG_COUNT: 1
GIT_CONFIG_KEY_0: url.https://${{ secrets.<token> }}@github.com.insteadOf
GIT_CONFIG_VALUE_0: ssh://git@github.com
POETRY_EXPERIMENTAL_SYSTEM_GIT_CLIENT: 1
```
(The last one is required as poetry uses [dulwich](https://python-poetry.org/docs/configuration/#experimentalsystem-git-client) by default, that does not recognize 
**GIT_CONFIG_{KEY|VALUE}** parameters)

@rarkins I wonder whether my request is not a duplicate of https://github.com/renovatebot/renovate/issues/11470?

FYI: given that we have a workaround, the priority of the feature is not high anymore for us.

**Reply by rarkins** [reply]
It's definitely related / could be the same solution
